File: 3541-FindMostFrequentVowelandConsonant/3541-FindMostFrequentVowelandConsonant.py
# Last updated: 9/13/2025, 8:55:41 PM
class Solution:
    def maxFreqSum(self, s: str) -> int:
        vowels = "aeiou"
        vowels_list = []
        consonant_list = []
        for c in s:
            if c in vowels:
                vowels_list.append(c)
            else:
                consonant_list.append(c)

        max_vowel_count = Counter(vowels_list).most_common(1)[0][1] if vowels_list else 0
        max_consonant_count = Counter(consonant_list).most_common(1)[0][1] if consonant_list else 0
        return max_vowel_count + max_consonant_count

File: 3541-FindMostFrequentVowelandConsonant/3541-FindMostFrequentVowelandConsonant.cpp
// Last updated: 9/13/2025, 9:21:50 PM
class Solution {
public:
    int maxFreqSum(string s) {
        int freq[26], max_vowel = 0, max_conso = 0;

        for (char c: s) {
            int i = c - 'a';
            freq[i] ++;

            if (c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u') {
                max_vowel = max(freq[i], max_vowel);
            } else {
                max_conso = max(freq[i], max_conso);
            }
        }

        return max_vowel + max_conso;
    }
};